backend/code/archipelag/market/views.py

```diff
@@ -67,16 +69,30 @@ def create(self, request):
                 create market or id of new market
         """
         image_fields = request.data
-        market_id = image_fields.get("market_id")
-        newest_market = Market.objects.filter(id=market_id).first()
+        fields = dict(image_path=request.FILES.get("file"),
+                      market_id=image_fields.get("market_id"))
+        try:
+            self.validate_image(fields)
+        except ValidationError as error:
```

Comment:
ValidationError puts errors as Response with 400 code. If you want to parse ValidationError manually - try http://www.django-rest-framework.org/api-guide/exceptions/#custom-exception-handling